src/tpm2/crypto/openssl/CryptEccSignature.c

```diff
@@ -616,6 +618,63 @@ BnValidateSignatureEcdsa(
  Exit:
     return retVal;
 }
+#else // USE_OPENSSL_FUNCTIONS_ECDSA     libtpms added begin
+TPM_RC
+BnValidateSignatureEcdsa(
+			 bigNum                   bnR,           // IN: r component of the signature
+			 bigNum                   bnS,           // IN: s component of the signature
+			 bigCurve                 E,             // IN: the curve used in the signature
+			 //     process
+			 bn_point_t              *ecQ,           // IN: the public point of the key
+			 const TPM2B_DIGEST      *digest         // IN: the digest that was signed
+			 )
+{
+    ECDSA_SIG        *sig = NULL;
+    EC_KEY           *eckey = NULL;
+    int               retVal;
+    int               rc;
+    BIG_INITIALIZED(r, bnR);
+    BIG_INITIALIZED(s, bnS);
+    EC_POINT         *q = EcPointInitialized(ecQ, E);
+
+    sig = ECDSA_SIG_new();
+    eckey = EC_KEY_new();
+
+    if (r == NULL || s == NULL || q == NULL ||
+        sig == NULL ||
+        eckey == NULL ||
+        EC_KEY_set_group(eckey, E->G) != 1 ||
+        EC_KEY_set_public_key(eckey, q) != 1 ||
+        ECDSA_SIG_set0(sig, r, s) != 1 /* keep last */)
+        ERROR_RETURN(TPM_RC_FAILURE);
+
+    /* ECDSA_SIG_set0 will cause r and s free() */
```

Comment:
This statment is incorrect, althoiugh the code handles it ok, the statement should be somethign like:
/* ECDSA_SIG_set0 takes ownership of r and s, prevent explicit free() */

Reply:
fixed